clients/src/test/java/org/apache/kafka/clients/InFlightRequestsTest.java

```diff
@@ -0,0 +1,188 @@
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class InFlightRequestsTest {
+
+    private final InFlightRequests inFlightRequests = new InFlightRequests(Integer.MAX_VALUE);
+    private final List<NetworkClient.InFlightRequest> requests = new ArrayList<>();
+    private final Map<String, Integer> nodeWiseRequestCounter = new HashMap<>();
+    private Integer nodeIdCounter = 0;
+
+    @Before
+    public void setUp() throws Exception {
+        for (nodeIdCounter = 0; nodeIdCounter < 2; nodeIdCounter++) {
+            String nodeId = nodeIdCounter.toString();
+            pushRequests(nodeId, 5);
+        }
+    }
+
+    @Test
+    public void testAdd() {
+        String nodeId = (nodeIdCounter++).toString();
+        int totalRequest = 10;
+        pushRequests(nodeId, totalRequest);
+        assertEquals(totalRequest, inFlightRequests.count(nodeId));
+    }
+
+    @Test
+    public void testCompleteNext() {
+        NetworkClient.InFlightRequest oldestRequest = requests.remove(0);
+        decrementCounter(oldestRequest.destination);
+        assertEquals(oldestRequest, inFlightRequests.completeNext(oldestRequest.destination));
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void testCompleteNextWithInvalidNode() {
+        inFlightRequests.completeNext("");
+    }
+
+
+    @Test
+    public void testLastSent() {
+        NetworkClient.InFlightRequest lastSentRequest = requests.get(requests.size() - 1);
+        assertEquals(lastSentRequest, inFlightRequests.lastSent(lastSentRequest.destination));
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void testLastSentWithInvalidNode() {
+        inFlightRequests.lastSent("");
+    }
+
+    @Test
+    public void testCompleteLastSent() {
+        NetworkClient.InFlightRequest lastSentRequest = requests.remove(requests.size() - 1);
+        decrementCounter(lastSentRequest.destination);
+        assertEquals(lastSentRequest, inFlightRequests.completeLastSent(lastSentRequest.destination));
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void testCompleteLastSentWithInvalidNode() {
+        inFlightRequests.completeLastSent("");
+    }
+
+    @Test
+    public void testCanSendMore() {
+    }
+
+    @Test
+    public void testCount() {
+        assertEquals(requests.size(), inFlightRequests.count());
+    }
+
+    @Test
+    public void testIsEmpty() {
+        assertEquals(inFlightRequests.count() == 0, inFlightRequests.isEmpty());
+    }
+
+    @Test
+    public void testCountInSingleNode() {
+        String nodeId = (nodeIdCounter++).toString();
+        int totalRequest = 10;
+        pushRequests(nodeId, totalRequest);
+        assertEquals(totalRequest, inFlightRequests.count(nodeId));
+    }
+
+    @Test
+    public void testIsEmptyInSingleNode()  {
+        String nodeId = (nodeIdCounter++).toString();
+        assertTrue(inFlightRequests.isEmpty(nodeId));
```

Comment:
May be worth adding an error message for `aasertTrue` (in all the places where assertTrue is used).